cloud/store/store.go

```diff
@@ -205,6 +220,35 @@ func (st *Store) Open(ctx context.Context) (err error) {
 		st.loadDatabase(ctx)
 	}
 
+	existedConfig, err := st.configViaPeers(filepath.Join(st.raftDir, peersFileName))
+	if err != nil {
+		return err
+	}
+
+	if servers, recover := st.recoverable(existedConfig.Servers); recover {
+		st.log.Info("recovery: start recovery with",
+			"peers", servers,
+			"snapshot_index", snapshotIndex(snapshotStore),
+			"last_applied_log_index", st.initialLastAppliedIndex)
+
+		if err := raft.RecoverCluster(st.raftConfig(), st, logCache, st.logStore, snapshotStore, st.transport, raft.Configuration{
+			Servers: servers,
+		}); err != nil {
+			return fmt.Errorf("raft recovery failed: %w", err)
+		}
+
+		// load the database if <= because RecoverCluster() will implicitly
+		// commits all entries in the previous Raft log before starting
+		if st.initialLastAppliedIndex <= snapshotIndex(snapshotStore) {
+			st.loadDatabase(ctx)
```

Comment:
I believe st.loadDatabase(ctx) in this case is not necessary and could also be wrong because it may interfere with the Raft workflow. The db is loaded when there is a consensus among the nodes what the latest update was

Reply:
raft.RecoverCluster method is commiting when succeeded, if the db is not loaded i think nothing would be applied to the node and that's why for example the replication was failing